File: java/src/main/java/TrainService.java
import java.util.ArrayList;
import java.util.List;

public class TrainService {
    private List<Ticket> tickets = new ArrayList<>();

    public void purchaseTicket(String from, String to, String userFirstName, String userLastName,
                                String userEmail, int price, String section) {
        Ticket ticket = new Ticket(from, to, userFirstName, userLastName, userEmail, price, section);
        tickets.add(ticket);
    }

    public Ticket getReceiptDetails(String userEmail) {
        for (Ticket ticket : tickets) {
            if (ticket.getUserEmail().equals(userEmail)) {
                return tickets;
            }
        }
        return null; // User not found
    }

    public List<String> getUsersBySection(String section) {
        List<String> userSeats = new ArrayList<>();
        for (Ticket ticket : tickets) {
            if (ticket.getSection().equals(section)) {
                userSeats.add(ticket.getUserFirstName() + " " + ticket.getUserLastName() + " - " + ticket.getSeat());
            }
        }
        return userSeats;
    }

    public void removeUser(String userEmail) {
        tickets.removeIf(ticket -> ticket.getUserEmail().equals(userEmail));
    }

    public void modifyUserSeat(String userEmail, String newSeat) {
        for (Ticket ticket : tickets) {
            if (ticket.getUserEmail().equals(userEmail)) {
                ticket.setSeat(newSeat);
                break;
            }
        }
    }
    public List<Ticket> getTickets() {
        return tickets;
    }
}

File: python/train_service.py
class TrainService:
    def __init__(self):
        self.tickets = []

    def purchase_ticket(self, from_location, to_location, user_first_name, user_last_name,
                        user_email, price, section):
        ticket = Ticket(from_location, to_location, user_first_name, user_last_name,
                        user_email, price, section)
        self.tickets.append(ticket)

    def get_receipt_details(self, user_email):
        for ticket in self.tickets:
            if ticket.user_email == user_email:
                return ticket
        return None  # User not found

    def get_users_by_section(self, section):
        user_seats = []
        for ticket in self.tickets:
            if ticket.section == section:
                user_seats.append(f"{ticket.user_first_name} {ticket.user_last_name} - {ticket.seat}")
        return user_seats

    def remove_user(self, user_email):
        self.tickets = [ticket for ticket in self.tickets if ticket.user_email != user_email]

    def modify_user_seat(self, user_email, new_seat):
        for ticket in self.tickets:
            if ticket.user_email == user_email:
                ticket.seat = new_seat
                break
